### DISCUSSION TITLE
cy.contains('xxx').should('not.exist') try to expect <body> not to exist in the DOM

### DISCUSSION THREAD

**Discussion by aNyMoRe0505** [cypress-io/cypress#14879] [Questions and Help]
I don't know If I can repost here, I have already posted an [issue](https://github.com/cypress-io/cypress/issues/14861)
Please let me know if I need to delete this discussion or not.

I try to test input validation by cypress
When user enter page first time, user should not see the error message
So I write test like this
`cy.contains('forgot_password_error_email_empty').should('not.exist');`
but it seems that command try to expect body not to exist in the DOM so it fails in the end

<img width="327" alt="2" src="https://user-images.githubusercontent.com/9814815/106564709-91ad7400-6568-11eb-84e8-3593970aa168.png">

When I reproduced my problem, I found this bug is related to `next-i18n` and only occur in `next production build`
If I have translation key `test` (public/static/locales/en) `cy.contains('test').should('not.exist')` will check body exist or not

If I write cy.contains with selector `cy.contains('span', 'test').should('not.exist')`, the test will pass.
But I think this is not the good solution.

### Test code to reproduce
clone this project (https://github.com/aNyMoRe0505/next-cypress)
npm install => npm run build =>  npm run start
then npm run cypress:open to execute test
### Versions
"cypress": "^6.3.0"

**Comment by dillibabuj** [comment]
Hey @aNyMoRe0505 
cy.contains() expects the element with content to eventually exist in the DOM. [Refer Default Assertions](https://docs.cypress.io/guides/core-concepts/introduction-to-cypress.html#For-instance)

You could rather do this : `cy.get('body').should('not.contain','text-you-expect-not-to-be-present')`
